presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -211,6 +212,11 @@ public CounterStat getHighMemorySplitSource()
 
     private Iterable<HivePartitionMetadata> getPartitionMetadata(SemiTransactionalHiveMetastore metastore, Table table, SchemaTableName tableName, List<HivePartition> hivePartitions, Optional<HiveBucketProperty> bucketProperty)
     {
+        String tableNotReadable = table.getParameters().get(OBJECT_NOT_READABLE);
+        if (!isNullOrEmpty(tableNotReadable)) {
+            throw new HiveNotReadableException(tableName, Optional.empty(), tableNotReadable);
```

Comment:
will the exception message include table name?

Reply:
@findepi : Fixed! Thank you !